packages/venia-concept/src/components/PurchaseDetailsPage/OrderItem/orderItem.js

```diff
@@ -1,12 +1,78 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { Price } from '@magento/peregrine';
+import classify from 'src/classify';
+import defaultClasses from './orderItem.css';
+import ButtonGroup from './ButtonGroup';
+import { itemPropType } from './constants';
+import { getButtonGroupItems } from './helpers';
 
-export default class OrderItem extends Component {
+class OrderItem extends Component {
     static propTypes = {
-        item: PropTypes.shape({})
+        classes: PropTypes.shape({
+            root: PropTypes.string,
+            main: PropTypes.string,
+            imageAndPropsContainer: PropTypes.string,
+            titleImage: PropTypes.string,
+            propsColumnContainer: PropTypes.string,
+            title: PropTypes.string,
+            priceContainer: PropTypes.string
+        }),
+        item: itemPropType,
+        onBuyAgain: PropTypes.func,
+        onShare: PropTypes.func,
+        currencyCode: PropTypes.string
+    };
+
+    //TODO: get currencyCode whether from item object or from order or from user cart
+    static defaultProps = {
+        currencyCode: 'USD'
+    };
+    //TODO: make correct mapping from item
+    onBuyAgainHandler = () => {
+        this.props.onBuyAgain({ item: this.props.item, quantity: 1 });
+    };
+    //TODO: make correct mapping from item
+    onShareHandler = () => {
+        this.props.onShare({ item: this.props.item, quantity: 1 });
```

Comment:
You do not need quantity here, this callback do not need it.

Reply:
Look at the signature of the handleShare method of PurchaseDetails component. It requires just an item object.

Reply:
You're right